### DISCUSSION TITLE
Publishing crates on crates.io

### DISCUSSION THREAD

**Discussion by trevor-crypto** [o1-labs/proof-systems#963] [General]
Since https://github.com/o1-labs/proof-systems/pull/842#issuecomment-1397496696 has been merged, it makes sense to publish crates to crates.io.

The main reason to do this is because when another project is using these crates, it is much easier to keep track of which version is being used and when to upgrade.

For example, I am working on a project that implements Mina support therefore we require some crates as dependencies. Right now, we need to add the crates to `Cargo.toml` as a git dependency with a specific revision. This is not really sustainable for us because we will be unaware when a crate is updated. If these crates are versioned properly, we will be alerted by dependabot that there is a new version and we can upgrade accordingly and make changes to our code to continue supporting Mina.

**Comment by mimoo** [comment]
Thanks for posting this! Can I ask what Mina project you're working on, or why you need to access kimchi directly? I believe most projects are developing on Mina using snarkyjs which is much more polished for external use. (I've been looking for an excuse to publish our crates so this is interesting.)

**Reply by trevor-crypto** [reply]
I don't really know much about kimchi, except that I like to eat it!

The use case is for basic wallet functionality with Rust - implementing singing and key generation for Mina, using proprietary signing mechanisms.

**Reply by mimoo** [reply]
Ah I see, so you're mostly using this crate then https://github.com/o1-labs/proof-systems/tree/master/signer ?

(cc @jspada )

**Reply by trevor-crypto** [reply]
Just these: mina-signer, mina-hasher, mina-curves, o1-utils
